src/os_sensor_nodelet.cpp

```diff
@@ -287,25 +293,44 @@ class OusterSensor : public OusterClientBase {
             config_flags |= ouster::sensor::CONFIG_UDP_DEST_AUTO;
         }
 
+        if (sensor::in_multicast(udp_dest)) {
+            if (is_arg_set(mtp_dest_arg)) {
+                NODELET_INFO("Will recieve data via multicast on %s", mtp_dest_arg.c_str());
+                this->mtp_dest = mtp_dest_arg;
+            } else {                
+                NODELET_INFO("mtp_dest arg not defined, will recieve data via multicast"
+                             "on first available interface");
+                this->mtp_dest = std::string{};
+            }
+
+            this->mtp_main = mtp_main_arg;
+        }
+
         return std::make_pair(config, config_flags);
     }
 
     void configure_sensor(const std::string& hostname,
```

Comment:
@ShepelIlya can you double check this suggestions? The current code breaks the case when the user is using `sensor.launch`, in that launch file the expectation is that user configurations will be applied even when `mtp_main` is not set. 
```suggestion
    void configure_sensor(const std::string& hostname,
                          sensor::sensor_config& config,
                          int config_flags) {
        if (sensor::in_multicast(udp_dest) && !mtp_main) {
            if (!get_config(hostname, config, true)) {
                NODELET_ERROR("Error getting active config");
            }
            NODELET_INFO_STREAM("Retrived active config of sensor");
            return;
        }

        try {
            if (!set_config(hostname, config, config_flags)) {
                auto err_msg = "Error connecting to sensor " + hostname;
                NODELET_ERROR_STREAM(err_msg);
                throw std::runtime_error(err_msg);
            }
        } catch (const std::exception& e) {
            NODELET_ERROR("Error setting config:  %s", e.what());
            throw;
        }

        NODELET_INFO_STREAM("Sensor " << hostname
                                      << " configured successfully");
    }
```

Reply:
also is it fair to assume that in a multicast configuration that `udp_dest` can not be empty for both the main and any secondary connection?

Reply:
Yes, you are right, I missed case with using default sensor.launch - fixed in  [d58f422](https://github.com/ouster-lidar/ouster-ros/pull/57/commits/d58f4229245f248de95ad32c4089cc5d2686f308).

Reply:
About `udp_dest` argument - of course, we can try to load this value from sensor with `get_config`, but i think that this will make configure api more complicated for no reason. With `udp_dest` as required value we know that we will use multicast client at `create_sensor_config_rosparams` stage, before any communication with sensor